src/main/java/com/influxdb/v3/client/internal/InfluxDBClientImpl.java

```diff
@@ -183,16 +185,48 @@ public Stream<Object[]> query(@Nonnull final String query,
         return queryData(query, parameters, options)
                 .flatMap(vector -> {
                     List<FieldVector> fieldVectors = vector.getFieldVectors();
-                    return IntStream
-                            .range(0, vector.getRowCount())
-                            .mapToObj(rowNumber -> {
-
-                                ArrayList<Object> row = new ArrayList<>();
-                                for (FieldVector fieldVector : fieldVectors) {
-                                    row.add(fieldVector.getObject(rowNumber));
-                                }
-                                return row.toArray();
-                            });
+                    var fields = vector.getSchema().getFields();
+                    return IntStream.range(0, vector.getRowCount())
+                           .mapToObj(rowNumber -> {
+                               ArrayList<Object> row = new ArrayList<>();
+                               for (int i = 0; i < fieldVectors.size(); i++) {
+                                   var schema = fields.get(i);
+                                   var metaType = schema.getMetadata().get("iox::column::type");
```

Comment:
Try to simplify like this:
```suggestion
                               for (FieldVector fieldVector : fieldVectors) {
                                   Field field = fieldVector.getField();
                                   var metaType = field.getMetadata().get("iox::column::type");
```